### PR TITLE
test: backfill unit tests for 7 untested migration components

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

Closes Cosmic Python audit finding #4 — 7 components in `src/application/components/` had zero unit-test coverage. Each gets a focused 3-5 test regression file covering the happy path plus at least one error/edge-case branch.

| Component | Tests | Coverage focus |
|-----------|-------|----------------|
| `admin_scheme_migration` | 4 | user+group role assignment, group refresh fallback, unmapped-project skip, OP failure propagation |
| `agile_board_migration` | 4 | board→query + sprint→version pipeline, missing project skip, closed-state mapping, Jira-side failure tolerance |
| `reporting_migration` | 4 | filter+dashboard happy path, dashboard fallback to reporting project, extract-failure short-circuit, OP failure counting |
| `tempo_account_migration` | 5 | extract on 200/non-200, name-match case insensitivity, dry-run placeholder, op_client invocation |
| `time_entry_migration` | 5 | Rails preflight, no-mapping skip, zero-created-with-input gating, success delegation, transformation-only ValueError |
| `work_package_content_migration` | 5 | mapping-missing hard-fail, jira_key→wp_id index build, bare-int legacy row skip, link rewrite, run() success |
| `work_package_skeleton_migration` | 5 | save persistence, save IO failure, incremental load, save-failed-with-creations hard-fail, zero-creations success |

**Total: 32 new tests.**

Tests follow the established `cfg.mappings` proxy monkeypatch pattern (per `tests/conftest.py` and existing tests like `test_affects_versions_migration.py`). No source code changes.

## Test plan

- [x] `ruff check tests/` passes (0 issues)
- [x] All 32 new tests pass
- [x] Full unit suite passes: 1540 passed, 56 warnings in 298s
- [x] No source code modified — tests only

### CODE REVIEW

File: tests/unit/test_time_entry_migration.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    tmp_path: Path,
+    monkeypatch: pytest.MonkeyPatch,
+    _mock_mappings: None,
+) -> None:
+    """Discovered>0 but migrated==0 → ComponentResult(success=False) per gating policy."""
+    # Seed a mapping file so the loader returns at least one WP.
+    mapping = {
+        "1001": {"jira_key": "PROJ-1", "openproject_id": 5001},
+    }
+    (tmp_path / "work_package_mapping.json").write_text(json.dumps(mapping))
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply tests/unit/test_time_entry_migration.py:101]: Addressed in [PR #235](https://github.com/netresearch/jira-to-openproject/pull/235) (commit 9f626c3). Added `encoding="utf-8"` to both `write_text` calls in this file. The atomic-temp-file rename suggestion intentionally not applied — tmp_path test fixtures have no concurrent readers, so the wrapper would only add boilerplate.

File: tests/unit/test_work_package_content_migration.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+def test_load_work_package_mapping_builds_jira_key_lookup(
+    tmp_path: Path,
+    _mock_mappings: None,
+) -> None:
+    """Dict-shaped rows populate the jira_key → wp_id index used by link rewriting."""
+    mapping = {
+        "1001": {"jira_key": "PROJ-1", "openproject_id": 5001},
+        "1002": {"jira_key": "PROJ-2", "openproject_id": 5002},
+    }
+    (tmp_path / "work_package_mapping.json").write_text(json.dumps(mapping))
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply tests/unit/test_work_package_content_migration.py:82]: Addressed in [PR #235](https://github.com/netresearch/jira-to-openproject/pull/235) (commit 9f626c3). All 4 `write_text` calls in this file now pass `encoding="utf-8"`. Same rationale as the time_entry comment for skipping the atomic-write wrapper in unit tests.

File: tests/unit/test_work_package_skeleton_migration.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    """_save_mapping writes the in-memory mapping to disk as JSON."""
+    mig = _build_mig(tmp_path)
+    mig.work_package_mapping = {
+        "1001": {"jira_key": "PROJ-1", "openproject_id": 5001, "project_key": "PROJ"},
+    }
+
+    assert mig._save_mapping() is True
+    assert mig._last_save_succeeded is True
+
+    on_disk = json.loads(mig.work_package_mapping_file.read_text())
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply tests/unit/test_work_package_skeleton_migration.py:68]: Addressed in [PR #235](https://github.com/netresearch/jira-to-openproject/pull/235) (commit 9f626c3). `read_text` now passes `encoding="utf-8"`.

File: tests/unit/test_work_package_skeleton_migration.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    )
+
+    seeded = {"1001": {"jira_key": "PROJ-1", "openproject_id": 5001}}
+    # Need to seed the file at the production data_dir location for the
+    # constructor to find it; do this by patching get_path before construction.
+    import src.config as cfg
+
+    monkeypatch_data_dir = tmp_path / "var_data"
+    monkeypatch_data_dir.mkdir()
+    (monkeypatch_data_dir / "work_package_mapping.json").write_text(json.dumps(seeded))
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply tests/unit/test_work_package_skeleton_migration.py:104]: Addressed in [PR #235](https://github.com/netresearch/jira-to-openproject/pull/235) (commit 9f626c3). `write_text` now passes `encoding="utf-8"`. Same rationale as the other comments for not applying the atomic-write wrapper in unit tests.

File: tests/unit/test_work_package_skeleton_migration.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+    original_get_path = cfg.get_path
+    cfg.get_path = lambda key: monkeypatch_data_dir if key == "data" else original_get_path(key)
+    try:
+        mig = WorkPackageSkeletonMigration(jira_client=MagicMock(), op_client=MagicMock())
+    finally:
+        cfg.get_path = original_get_path
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply tests/unit/test_work_package_skeleton_migration.py:111]: Addressed in [PR #235](https://github.com/netresearch/jira-to-openproject/pull/235) (commit 9f626c3). `test_load_existing_mapping_for_incremental_run` now uses `monkeypatch.setattr(cfg, "get_path", ...)` instead of manual try/finally — restoration is automatic on teardown, even when the test fails.

File: tests/unit/test_reporting_migration.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+    extracted = mig._extract()
+    mapped = mig._map(extracted)
+
+    assert mapped.success is True
+    # Reporting project ensured at least once.
+    assert len(op.reporting_calls) == 1
+    # Wiki payload built using fallback reporting project id (555).
+    assert mapped.data["wiki_pages"][0]["project_id"] == 555
+    assert mapped.details["skipped_dashboards"] == 0
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply tests/unit/test_reporting_migration.py:141]: Addressed in [PR #235](https://github.com/netresearch/jira-to-openproject/pull/235) (commit 9f626c3). The test now clears `cfg.openproject_config["reporting_wiki_project"]` and `["reporting_wiki_project_name"]` via `monkeypatch.setitem` before calling `_map()`, so a developer `.env.local` config cannot override the ensure_reporting_project fallback the assertion depends on.

File: tests/unit/test_time_entry_migration.py [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+    op = MagicMock()
+    if rails_present:
+        op.rails_client = MagicMock()
+    else:
+        # Explicitly set to None so getattr(..., "rails_client", None) returns None.
+        op.rails_client = None
+
+    mig = tem.TimeEntryMigration(jira_client=jira, op_client=op)
+    mig.data_dir = tmp_path
+    return mig
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply tests/unit/test_time_entry_migration.py:59]: Addressed in [PR #235](https://github.com/netresearch/jira-to-openproject/pull/235) (commit 9f626c3). `_make_mig` now `monkeypatch.setitem(cfg.jira_config, "projects", [])` so a developer with `J2O_JIRA_PROJECTS` set locally cannot drop the seeded `PROJ-1` mapping and route the test through the no_migrated_work_packages skip path.
